sourcegraph/sourcegraph.proto

```diff
@@ -588,6 +588,214 @@ service Repos {
 	rpc ListCommitters(ReposListCommittersOp) returns (CommitterList);
 }
 
+// StorageError represents an error when interacting with the Storage service.
+message StorageError {
+	// Code represents the type of error for programatic handling.
+	enum Code {
+		// None is the error code used when programatic handling of the error is not
+		// advised. It can be assumed that the operation failed, and that a human
+		// readable message is also provided.
+		None = 0;
+
+		// EOF is the error returned by Read when no more input is available.
+		// Functions should return EOF only to signal a graceful end of input. If
+		// the EOF occurs unexpectedly in a structured data stream, the appropriate
+		// error is either UNEXPECTED_EOF or some other error message giving more
+		// detail.
+		EOF = 1;
+
+		// NotExist is the error used when attempting to read/write to an object
+		// that does not exist.
+		NotExist = 2;
+
+		// Permission is the error used when permission to access the given file is
+		// not granted to you.
+		Permission = 3;
+	}
+
+	// code is the error code. If no error code is specified then programatic
+	// handling of the error is not advised. The user should be informed of the
+	// error message instead.
+	Code code = 1;
+
+	// message is the human-readable error message.
+	string message = 2;
+}
+
+// StorageName is a storage object's name.
+message StorageName {
+	// AppName is the name of the application whose data you are trying to
+	// read/write, applications may read and write to eachothers data assuming
+	// the admin has not restricted such access.
+	string app_name = 1;
+
+	// If specified storage is considered local to the given repository. Otherwise
+	// it is considered "global" (i.e. shared across all repositories).
+	string repo = 2;
```

Comment:
`Repo ...`.

I don't know how strongly we should enforce these, but the rest of this repo follows this catchable-with-`golint` style.